### PR TITLE
chore: refactor OfficialsList.tsx

### PR DESCRIPTION
**Pull request by jonahduckworth**
Extracted every component and util from `OfficialsList.tsx`

The goal is to clean up huge files to set ourselves up to make better use of state management and remote data
- With this, we want to ensure that while we waiting for a response, our remote data state will be `pending` and show the loading screen
- We might also be able to optimistically send API requests and update the front end right away (I'm thinking specifically for when assigning someone to a game)

### CODE REVIEW

File: src/components/OfficialsList/components/expandedDetails/DarkDaysCard.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          <td className="w-16">Time</td>
+          <td className="pl-4">Notes</td>
+        </tr>
+      </thead>
+      <tbody>
+        {blockedOffTimes &&
+          Object.keys(blockedOffTimes)
+            .filter(
+              (dateKey) =>
+                Utils.extractMonthYear(dateKey, currentDate) ===
```
Comment (rymccue) [inline src/components/OfficialsList/components/expandedDetails/DarkDaysCard.tsx:21]: Why are we filtering here?

File: src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <div className="flex items-center">
+        <p
+        className={`text-sm font-normal ${
+            assignedGamesAlready.length > 0
+            ? 'text-error-500'
+            : 'text-gray-900'
+        }`}
+        >
+        <strong>
+            @ {assignedGamesAlready[0].home_team.abbreviation}
```
Comment (rymccue) [inline src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx:13]: There will always be an item in this array?

File: src/components/OfficialsList/utils/getListOfSpecificRole.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const getListOfSpecificRole = (officialsArray: any, showReferees: boolean, showLinesmen: boolean, role: string) => {
+    return officialsArray.filter(
+        (official) =>
+            (showReferees && official.role?.Referee) ||
+            (showLinesmen && official.role?.Linesman) ||
+            role === 'supervisor',
```
Comment (rymccue) [inline src/components/OfficialsList/utils/getListOfSpecificRole.ts:6]: We always want to show supervisors?

File: src/components/OfficialsList/utils/getOfficialCalendarData.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        }
+
+        const blockedOffDates = officialsCalendarData[uid].blockedOffTimes;
+        const currentSelectedDate = parseISO(game.time);
+        const formattedTime = formatDate(currentSelectedDate);
+
+        try {
+            return blockedOffDates[formattedTime];
+        } catch (error) {
+            return null;
```
Comment (rymccue) [inline src/components/OfficialsList/utils/getOfficialCalendarData.ts:16]: Is it worth adding a log here so we know what's up if this fails?

File: src/components/OfficialsList/OfficialsList.tsx [hunk 5]
```diff
@@ -68,394 +51,69 @@ const OfficialsList = ({ game, role, isAssigned, close = () => {} }) => {
...
-  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setSearchTerm(e.target.value);
-    filterOfficials(showReferees, showLinesmen);
-  };
-
   const handleRefereeCheckboxChange = (
     e: React.ChangeEvent<HTMLInputElement>,
   ) => {
     setShowReferees(e.target.checked);
-    filterOfficials(e.target.checked, showLinesmen); // pass the updated state
```
[the remark this answers is not in the document]
Comment (jonahduckworth) [reply src/components/OfficialsList/OfficialsList.tsx:137]: Correct. Most of the logic is in `Utils.filterAndSortOfficials`

File: src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx (hunk 2 above)
Comment (jonahduckworth) [reply src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx:13]: Screenshot 2024-02-15 at 1 08 06 PM

yes

File: src/components/OfficialsList/utils/getOfficialCalendarData.ts (hunk 4 above)
Comment (jonahduckworth) [reply src/components/OfficialsList/utils/getOfficialCalendarData.ts:17]: [c2b88dc](https://github.com/RefBuddy/ref-buddy-web/pull/69/commits/c2b88dcc7abdd9f71dfa5e894650f72efd7d8667)

File: src/components/OfficialsList/components/expandedDetails/DarkDaysCard.tsx (hunk 1 above)
Comment (jonahduckworth) [reply src/components/OfficialsList/components/expandedDetails/DarkDaysCard.tsx:22]: [c2b88dc](https://github.com/RefBuddy/ref-buddy-web/pull/69/commits/c2b88dcc7abdd9f71dfa5e894650f72efd7d8667)

Comment (rymccue) [reply src/components/OfficialsList/components/expandedDetails/DarkDaysCard.tsx:22]: Ohhh, thanks for the context!

File: src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx (hunk 2 above)
Comment (rymccue) [reply src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx:13]: Is it worth passing that individual game in maybe? We only ever need that first item in the array in this component?

Comment (jonahduckworth) [reply src/components/OfficialsList/components/mainDetails/AssignedGameCard.tsx]: I see. Sick identification!

[6bed03e](https://github.com/RefBuddy/ref-buddy-web/pull/69/commits/6bed03ec90623c9d8914bdc321221bb2dbbb2f7c)

**Review by rymccue — APPROVED** [review]

**Pull request merged**